src/test/components/MenuButtons.test.js

```diff
@@ -209,26 +218,27 @@ describe('app/MenuButtons', function() {
       expect(queryPreferenceCheckbox()).toBeFalsy();
     });
 
-    it('can publish, cancel, and then publish again', () => {
+    it('can publish, cancel, and then publish again', async () => {
       const { profile } = createSimpleProfile();
       const {
         getPanel,
         getPublishButton,
         getCancelButton,
         getPanelForm,
-        resolveUpload,
         clickAndRunTimers,
       } = setup(profile);
       clickAndRunTimers(getPublishButton());
       fireEvent.submit(getPanelForm());
-      resolveUpload();
 
       // These shouldn't exist anymore.
       expect(() => getPanel()).toThrow();
       expect(() => getPublishButton()).toThrow();
 
       clickAndRunTimers(getCancelButton());
 
+      await wait(() => {
+        getPublishButton();
+      });
```

Comment:
now there's some asynchronicity between the moment the user presses cancel and the moment there's some action, because there's a promise being rejected and only then the dispatching is done.